docs/javascript-api.md

```diff
@@ -0,0 +1,2378 @@
+
+ > await confluxWeb.cfx.getTransactionReceipt('0xd1d5b66b0d3d215f042f3e4907a3bf0acf5984972527fbca45eb67286835260d');
+ {
+   status: true,
+   contractCreated: '0x079352147ce2de227af6fa963f603a35aed8e601',
+   ...
+ }
+
+ > contract.address = '0x079352147ce2de227af6fa963f603a35aed8e601'; // after deploy, you should add address to contract
+```
+
+## ConfluxWeb.cfx.Contract.clone
+
+Clones the current contract instance.
+
+`function clone()`
+
+### Parameters
+
+`void`
+
+### Return
+
+`Object` The new contract instance.
+
+### Example
+
+```js
+ > const clone = contract.clone()
+ > contract.defaultGas === clone.defaultGas;
+ true
+
+ > contract.defaultGasPrice === clone.defaultGasPrice;
+ true
+
+ > contract.address === clone.address; // only address difference, and clone one address to be undefined
+ false
+```
+----------
+# #ConfluxWeb.cfx
+
+
+
+## ConfluxWeb.cfx.defaultAccount
+
+This default address is used as the default "from" property.
+
+20 Bytes: Any Conflux address. You should have the private key for that address in your node or keystore.
+
+`string|undefined`
+
+### Example
+
+```js
+ > confluxWeb.cfx.defaultAccount
+ undefined
+
+ > confluxWeb.cfx.defaultAccount = '0xbbd9e9be525ab967e633bcdaeac8bd5723ed4d6b';
+ > confluxWeb.cfx.defaultAccount
+ "0xbbd9e9be525ab967e633bcdaeac8bd5723ed4d6b"
+```
+
+## ConfluxWeb.cfx.defaultEpoch
+
+When requests are made that act on the state of conflux,
+the default epoch parameter determines the height of the epoch.
+
+The following options are possible for the default epoch parameter:
+
+- `number`: An integer epoch number.
+- `"earliest"`: The earliest epoch where the genesis block in.
+- `"latest_state"`: The latest epoch where the latest block with an executed state in. (default)
+- `"latest_mined"`: The latest epoch where the latest mined block in.
+
+`number|string`
+
+### Example
+
+```js
+ > confluxWeb.cfx.defaultEpoch; // Default is "latest_state"
+ "latest_state"
+```
+
+## ConfluxWeb.cfx.currentProvider
+
+Will return the current provider, otherwise null
+
+`object|null`
+
+### Example
+
+```js
+ > confluxWeb.cfx.currentProvider
+ HttpProvider {
+  host: 'http://testnet-jsonrpc.conflux-chain.org:12537',
+  ...
+ }
+```
+
+## ConfluxWeb.cfx.setProvider
+
+Will change the provider for its module.
+
+`function setProvider(cfxProvider)`
+
+### Parameters
+
+Name        | Type   | Required | Default | Description
+------------|--------|----------|---------|------------------
+cfxProvider | object | true     |         | A valid provider.
+
+### Return
+
+`boolean` 
+
+### Example
+
+```js
+ > confluxWeb.currentProvider.constructor.name
+ "HttpProvider"
+ > confluxWeb.currentProvider.constructor.name
+ "HttpProvider"
+ > confluxWeb.cfx.setProvider('ws://localhost:8546') // change module provider
+ > confluxWeb.currentProvider.constructor.name
+ "HttpProvider"
+ > confluxWeb.currentProvider.constructor.name
+ "WebsocketProvider"
```

Comment:
why it called confluxWeb.currentProvider.constructor.name twice and return different values?
Also, don't use WebSocket as example